docs/csharp/versioning.md

```diff
@@ -0,0 +1,128 @@
+* `MINOR` is incremented when you add functionality in a backwards-compatible manner
+* `PATCH` is incremented when you make backwards-compatible bug fixes
+
+There are also ways to specify other scenarios like pre-release versions etc. Go [here](http://semver.org/) to get more information on SemVer
+and how best to utilize it when applying version information to your .NET library.
+
+### Application Configuration File
+
+As a .NET developer there's a very high chance you've encountered the `app.config` file present in most project types.
+This simple configuration file can go a long way into improving the rollout of new updates. You should generally design your libraries in such
+a way that information that is likely to change regularly is stored in the `app.config` file, this way when such information is updated
+the config file of older versions just needs to be replaced with the new one without the need for recompilation of the library.
+Go [here](https://msdn.microsoft.com/en-us/library/1fk1t1t0(v=vs.110).aspx) to get more information on the structure of the configuration file
+and what kind of information is usually stored in it.
+
+### Backwards Compatibility
+
+As you release new versions of your library, backwards compatibility with previous versions will most likely be one of your major concerns.
+A new version of your library is source compatible with a previous version if code that depends on the previous version can, when recompiled, work with the new version. 
+In contrast, a new version of your library is binary compatible if an application that depended on the old version can, without recompilation, work with the new version.
+
+Here are some things to consider when trying to maintain backwards compatibility with older versions of your library:
+
+* Virtual methods: When you make a virtual method non-virtual in your new version it means that projects that override that method
+will have to be updated.
+* Method signatures: When updating a method behaviour requires you to change it's signature as well, you should instead create an override so that derived classes calling into that method will still work.
+You can always manipulate the old method signature to call into the new method signature so that implementation remains consistent.
+* Optional Method Arguments: When you make previously optional method arguments compulsory then all code that does not supply those arguments will need to be updated.
+On the other hand making compulsory arguments optional should have very little effect especially if it doesn't change the method's behaviour.
+* Obsolete attribute: You can use this attribute in your code to specify classes or class members that are deprecated and likely to be removed in future versions.
+This ensures developers utilizing your library are better prepared for breaking changes. More info on the `Obsolete` attribute [here](https://msdn.microsoft.com/en-us/library/22kk2b44(v=vs.90).aspx).
```

Comment:
I think this should link to docs.MS instead of MSDN. And I would avoid using `[here]` as well. The two changes put together:

```md
* [Obsolete attribute](programming-guide/concepts/attributes/common-attributes.md#Obsolete): You can use this attribute …
```